src/textual/app.py

```diff
@@ -1945,6 +1945,29 @@ def push_screen(
         else:
             return await_mount
 
+    @overload
+    async def push_screen_wait(
+        self, screen: Screen[ScreenResultType]
+    ) -> ScreenResultType:
+        ...
+
+    @overload
+    async def push_screen_wait(self, screen: str) -> Any:
+        ...
+
+    async def push_screen_wait(
+        self, screen: Screen[ScreenResultType] | str
+    ) -> ScreenResultType:
```

Comment:
I don't understand why `-> ScreenResultType | Any` isn't needed.

Reply:
I think you're right. I don't know why it didn't complain.